test/wallaby/browser/local_storage_test.exs

```diff
@@ -16,24 +16,32 @@ defmodule Wallaby.Browser.LocalStorageTest do
     |> visit("index.html")
     |> execute_script(@set_value_script)
 
-    assert session
-    |> execute_script(@get_value_script) == "foo"
+    session
+    |> execute_script(@get_value_script, fn(value) -> send self(), {:result, value} end)
+
+    assert_received {:result, "foo"}
```

Comment:
I couldn't guarantee that the assert was being called in the callback. If I removed the code for calling the callback in execute_script, the assert isn't checked
``` Elixir
  def execute_script(session, script, callback) when is_function(callback) do
    {:ok, value} = Driver.execute_script(session, script, [])
    #callback.(value)  -- don't execute the callback
    session


-- In test: 

  result =
      session
      |> visit("page_1.html")
      |> execute_script(@script, ["now you see me", "return value"], fn(value) ->
           assert value == "return value"  #-- this still passes, is never actually asserted
         end)
  end
```